nvflare/app_opt/sklearn/sklearn_executor.py

```diff
@@ -23,44 +23,98 @@
 from nvflare.apis.fl_context import FLContext
 from nvflare.apis.shareable import Shareable, make_reply
 from nvflare.apis.signal import Signal
+from nvflare.app_common.abstract.learner_spec import Learner
 from nvflare.app_common.app_constant import AppConstants
 from nvflare.security.logging import secure_format_exception
 
 
+def _get_global_params(shareable: Shareable, fl_ctx: FLContext):
+    # retrieve current global params download from server's shareable
+    dxo = from_shareable(shareable)
+    current_round = shareable.get_header(AppConstants.CURRENT_ROUND)
+    fl_ctx.set_prop(AppConstants.CURRENT_ROUND, current_round)
+    return current_round, dxo.data
+
+
 class SKLearnExecutor(Executor):
-    def __init__(self, learner_id: str):
+    def __init__(self, learner_id: str, train_task=AppConstants.TASK_TRAIN):
+        """An Executor interface for scikit-learn Learner.
+
+        Args:
+            learner_id (str): id pointing to the learner object
+            train_task (str, optional): label to dispatch train task. Defaults to AppConstants.TASK_TRAIN.
+        """
         super().__init__()
-        self.client_id = None
-        self.writer = None
         self.learner_id = learner_id
+        self.learner = None
+        self.train_task = train_task
         self.local_model_path = None
         self.global_model_path = None
-        self.learner = None
+        self.client_id = None
+        self.writer = None
         self.fl_ctx = None
```

Comment:
Do we need a self.fl_ctx? It seems like this could be easily out of sync with the actual context.

Reply:
Very good point, I will address this in the next refactoring PR